### PR TITLE
feat(related-documents): add purple theme and adjust rendering logic when there's no post

### PR DESCRIPTION
**Pull request by sahmad-merative**
## Issue

Fixes #<gh-issue-id> [MERATIVE-738](https://jira.wh-sdlc.watson-health.ibm.com/browse/MERATIVE-738)

## Description

- Added functionality to add cards color to change to purple when `purple-cards` is added to the component.
- If no information is added to the component linkable-cards component will automatically fetch info from [sharepoint](https://merative.sharepoint.com/:x:/s/Merative.comWebsite/EQKpwTsG5bxPujqa635th6oB5u9HTRegXQ8h4VseREkgnA?e=AKPAsm)

## Design Specs

- Figma Link -https://www.figma.com/file/dFJwrp6VkUMSirF2KMLRMx/Thought-leadership-library?type=design&node-id=357%3A17984&t=iaNohDrbt4220FLs-1

## Test URLs
  
- Before (Changes from `main`): https://main--merative2--sahmad-merative.hlx.page/drafts/saad/leadspace-test
- After (Changes from this PR): https://194-linkable-cards--merative2--sahmad-merative.hlx.page/drafts/saad/leadspace-test
  
## Testing Instruction

- To test card's color `purple-cards` have to be added to the component as shown in the imae below
image

- If no information is added to the component linkable-cards component will automatically fetch info from [sharepoint](https://merative.sharepoint.com/:x:/s/Merative.comWebsite/EQKpwTsG5bxPujqa635th6oB5u9HTRegXQ8h4VseREkgnA?e=AKPAsm) and populate 3 cards from the list

-

### CODE REVIEW

File: blocks/linkable-cards/linkable-cards.css [hunk 1]
```diff
@@ -66,12 +66,26 @@
   box-shadow: 2px 4px 4px 2px rgb(0 0 0 / 15%);
 }
 
+.linkable-cards.purple-cards > div {
+  background-color: #6400A6;
```
Comment (proeung) [inline blocks/linkable-cards/linkable-cards.css:70]: Let's use a CSS variable for this hex code. You can add this `purple-140` to the main stylesheet. (https://github.com/hlxsites/merative2/blob/main/styles/styles.css#L93) 

`--purple-140: #6400A6;`


```suggestion
  background-color: var(--purple-140);
```

**Review by helms-charity — DISMISSED** [review]

File: blocks/linkable-cards/linkable-cards.js [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+    const json = await resp.json();
+    [...json.data].forEach((element, index) => {
+      if (index < 3) {
+        html += `<div><a href="e${element.path}"><p>${element.parent}</p>
+        <h3 id="for-contract-research-organizations">${element.title}</h3>
+        <p>${element.description}</p></a></div>`;
+        block.innerHTML = html;
+      }
+    });
+  }
```
Comment (proeung) [inline blocks/linkable-cards/linkable-cards.js:22]: We'll need to move this `else` fetching method to the `Related Resources` block (https://github.com/hlxsites/merative2/blob/main/blocks/related-resources/related-resources.js) as this is what it's being used on the existing Case Study and other doc pages (see - [sharepoint doc](https://merative.sharepoint.com/:w:/r/sites/Merative.comWebsite/_layouts/15/Doc.aspx?sourcedoc=%7B4D3CE5C4-9E61-4328-BD14-293C79B7347D%7D&file=4-ways-case-study.docx&action=default&mobileredirect=true)).

The goal is to display the latest 3 posts from the document query index when this block is empty. Also, we should query from the same `assettype` value and not just grab the top 3 items from the query index (https://main--merative2--hlxsites.hlx.page/documents/query-index.json).

Screen Shot 2023-06-22 at 10 33 58 AM

**Review by proeung — CHANGES_REQUESTED** [review]
@sahmad-merative Looks good! I just have a couple of comments/feedback. Also, thanks for following the PR template!

**Feedback/reminder for the next PR**

- I noticed that you're not using [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#examples). This might be because you haven't updated your package dependencies, which includes changes from this PR (https://github.com/hlxsites/merative2/pull/212/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R39-R40).. Please make sure to run `npm install` in your local build.

**Comment by proeung** [comment]
@sahmad-merative Any update on this PR? It's been over a week (9/22) since I left my review and I have not seen any follow commits to address and finish up this work.

cc: @keith-kaplan @Shalini-SB

File: blocks/linkable-cards/linkable-cards.js (hunk 2 above)
Comment (sahmad-merative) [reply blocks/linkable-cards/linkable-cards.js]: Done

File: blocks/linkable-cards/linkable-cards.css (hunk 1 above)
Comment (sahmad-merative) [reply blocks/linkable-cards/linkable-cards.css]: Done

**Comment by sharathmrft** [comment]
Issues/observations
 
1.Card height and width are not matching as per the design
2.No correct spacing between Linkable cards and Related documents
3.'Related Documents' section needs to be included in the design
4.No designs available for tab and Mobile
5.Not able to test the latest Modified document whether it is fetching latest 3 or not from the SharePoint
6.Background color of the component is not rendering in the Test link. each card background color only
  displaying- as per the design

@sahmad-merative @proeung

**Review by proeung — APPROVED** [review]
@sahmad-merative Looks good! Thanks for addressing my feedback.

I just checked a draft page of the "Case Study/Document" without links/content populated in the "Related Documents" block and I can see that the logic for fetching the latest content from the documents query index is showing up correctly (see attached).

https://194-linkable-cards--merative2--sahmad-merative.hlx.page/drafts/putra/4-ways-case-study

Screen Shot 2023-07-06 at 11 23 47 AM

I think we can merge the changes from this PR down to the `feat/thought-leadership` branch since the bulk of the changes is in place. As for @sharathmrft's feedback, we can revisit it when we conduct a holistic browser QA once all of the changes are merged into `feat/thought-leadership` branch (https://feat-thought-leadership--merative2--hlxsites.hlx.page/).

**Pull request merged**